pkg/loki/modules.go

```diff
@@ -156,7 +157,7 @@ func (t *Loki) initDistributor() (services.Service, error) {
 
 func (t *Loki) initQuerier() (services.Service, error) {
 	level.Debug(util.Logger).Log("msg", "initializing querier worker", "config", fmt.Sprintf("%+v", t.cfg.Worker))
-	worker, err := frontend.NewWorker(t.cfg.Worker, httpgrpc_server.NewServer(t.server.HTTPServer.Handler), util.Logger)
+	worker, err := frontend.NewWorker(t.cfg.Worker, cortex_querier.Config{MaxConcurrent: t.cfg.Querier.MaxConcurrent}, httpgrpc_server.NewServer(t.server.HTTPServer.Handler), util.Logger)
```

Comment:
@joe-elliott So this is new, I did a PR a long long time ago to make the frontend and the worker cortex agnostic and this seems to bring back dependency on cortex.

The problem I have is the fact that we expose in the worker the whole querier config while it seems we need only `MaxConcurrent`. 

I suggest we keep it like that for now in Loki, but we change cortex to take an interface so the contract is clear. WDYT ? I can send the PR.